sections/Content.js

```diff
@@ -0,0 +1,126 @@
+import { useState, useEffect, useRef } from 'react';
+import { supabase } from '../utilities/supabase';
+
+import { ThreeDots  } from  'react-loader-spinner'
+
+import { ExpressionCard } from '../components/container/index/expressioncard';
+
+import styles from '../styles/container/content.module.scss';
+
+function Content( { ssrData, posts } ) {
+
+    const HowMany = posts;
+
+    const [loading, setLoading] = useState(true);
+    const [expressions, setExpressions] = useState([]);
+    const [hasMore, setHasMore] = useState(true);
+
+    const [firstRender, setFirstRender] = useState(false);
+
+    const [from, setFrom] = useState(0);
+    const [to, setTo] = useState(HowMany-1);
+  
+    const elementRef = useRef(null);
+  
+    function onIntersection(entries) {
+        entries.forEach((entry) => {
+            if (entry.isIntersecting && hasMore) {
+                setFrom((from) => from + HowMany);
+                setTo((to) => to + HowMany);
+            }
+        });
+    }
+  
+    useEffect(() => {
+        const observer = new IntersectionObserver(onIntersection);
+
+        if (observer && elementRef.current && hasMore ) {
+        observer.observe(elementRef.current);
+        }
+
+        return () => {
+            if (observer) {
+                observer.disconnect();
+            }
+        };
+    }, [elementRef.current]);  
+    
+    useEffect(() => {
+        if(!firstRender) {
+            setFirstRender(true);
+        } else {
+            getData();
+        }
+    }, [from, to]);
+
+    const delay = ms => new Promise(
+        resolve => setTimeout(resolve, ms)
+    );
+
+    async function getData() {
+        if (hasMore) {
+            setLoading(true);
+        }
+
+        await delay(500);
```

Comment:
De ce pui artificial delay?

Reply:
Am zis că e mai interesant dacă pare că se incarcă ceva. Este o practică greșită?

Reply:
E mai bine să apară loading doar când e nevoie.